nodejs/nodejs-cloud-run-hello-world/app.js

```diff
@@ -0,0 +1,54 @@
+const express = require('express');
+const handlebars = require('handlebars');
+const fs = require('fs').promises;
+const metadata = require('gcp-metadata');
+
+const app = express();
+// Serve the files in /assets at the URI /assets.
+app.use('/assets', express.static('assets'));
+
+// The HTML content is produced by rendering a handlebars template.
+// The template values are stored in global state for reuse.
+const data = {
+  project: process.env.GOOGLE_CLOUD_PROJECT,
+  service: process.env.K_SERVICE || '???',
+  revision: process.env.K_REVISION || '???',
+};
+let template;
+
+app.get('/', async (req, res) => {
+  // The handlebars template is stored in global state so this will only once.
+  if (!template) {
+    // Load Handlebars template from filesystem and compile for use.
+    try {
+      const source = await fs.readFile('index.html.hbs', 'utf-8');
```

Comment:
nit, but since this resource is only needs to be loaded once, I might just:

```js
const template = handlebars.compile(fs.readFileSync('index.html.hbs', 'utf8'));
```

eliminates a little bit of boilerplate, if this is your only fs access, you could then:

```
const {readFileSync} = require('fs');
```

rather than pulling in `fs.promises`.